itests/clients/node_client.go

```diff
@@ -288,15 +291,89 @@ func (c *NodesClients) Handshake() {
 	c.ScalaClient.Connection.SendHandshake()
 }
 
-func (c *NodesClients) SendToNodes(t *testing.T, m proto.Message, scala bool) {
+func compactAndSort(s []Implementation) []Implementation {
+	byteSlice := make([]byte, len(s))
+	for i := range s {
+		byteSlice[i] = byte(s[i])
+	}
+
+	slices.Sort(byteSlice)
+	compacted := slices.Compact(byteSlice)
+
+	result := make([]Implementation, len(compacted))
+	for i := range compacted {
+		result[i] = Implementation(compacted[i])
+	}
+
+	return result
+}
+
+func (c *NodesClients) SendToScalaNode(t *testing.T, m proto.Message) {
+	t.Logf("Sending message to Scala node: %T", m)
+	c.ScalaClient.Connection.SendMessage(m)
+	t.Log("Message sent to Scala node")
+}
+
+func (c *NodesClients) SendToGoNode(t *testing.T, m proto.Message) {
 	t.Logf("Sending message to Go node: %T", m)
 	c.GoClient.Connection.SendMessage(m)
 	t.Log("Message sent to Go node")
-	if scala {
-		t.Logf("Sending message to Scala node: %T", m)
-		c.ScalaClient.Connection.SendMessage(m)
-		t.Log("Message sent to Scala node")
+}
+
+func (c *NodesClients) SendToNodes(t *testing.T, m proto.Message, nodes []Implementation) {
+	resultNodes := compactAndSort(nodes)
+	for i := range resultNodes {
+		switch resultNodes[i].String() {
+		case "Go":
+			c.SendToGoNode(t, m)
+		case "Scala":
+			c.SendToScalaNode(t, m)
+		default:
+			t.Fatalf("Node name is incorrect: %s", resultNodes[i].String())
+		}
+	}
```

Comment:
NodeGo and NodeScala constants can be used here
```suggestion
	ns := deduplicateImplementations(nodes)
	for i := range ns {
		switch ns[i] {
		case NodeGo:
			c.SendToGoNode(t, m)
		case NodeScala:
			c.SendToScalaNode(t, m)
		default:
			t.Fatalf("Unexpected node implementation %d", ns[i])
		}
	}
```
Also, better use shorter names for short live variables.
And clear error message suggested.